client/packages/system/src/Manage/Preferences/EditPage/usePreferenceSearch.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMemo, useState } from 'react';
+import { LocaleKey, useTranslation } from '@openmsupply-client/common';
+import { AdminPreferenceFragment } from '../api/operations.generated';
+import { PREFERENCE_GROUP_CONFIG } from './utils';
+
+export const usePreferenceSearch = (preferences: AdminPreferenceFragment[]) => {
+  const t = useTranslation();
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const filteredPreferences = useMemo(() => {
+    if (!searchTerm.trim()) {
+      return preferences;
+    }
+
+    const lowerSearch = searchTerm.toLowerCase();
+
+    return preferences.filter(pref => {
+      const translatedLabel = t(`preference.${pref.key}` as LocaleKey);
```

Comment:
these two type casts should really be avoided. 

however I'd rather not implement an enum or limited set of locale keys which would be needed to avoid the cast so let's roll with it